app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb

```diff
@@ -0,0 +1,124 @@
+          @extract_token_from_credentials = Authentication::AuthnJwt::InputValidation::ExtractTokenFromCredentials.new
+          @validate_and_decode_token_class = Authentication::AuthnJwt::ValidateAndDecode::ValidateAndDecodeToken
+          @identity_provider_factory = Authentication::AuthnJwt::IdentityProviders::CreateIdentityProvider
+          @validate_resource_restrictions_class = Authentication::ResourceRestrictions::ValidateResourceRestrictions
+          @create_identity_provider_class = Authentication::AuthnJwt::IdentityProviders::CreateIdentityProvider
+          @logger = Rails.logger
+        end
+
+        def authentication_parameters(authenticator_input)
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CREATING_AUTHENTICATION_PARAMETERS_OBJECT.new)
+          jwt_token = @extract_token_from_credentials.call(
+            credentials: authenticator_input.request.body.read
+          )
+          @authentication_parameters_class.new(
+            authentication_input: authenticator_input,
+            jwt_token: jwt_token
+          )
+        end
+
+        def jwt_identity(authentication_parameters)
+          identity_provider = @create_identity_provider_class.new.call(
+            authentication_parameters: authentication_parameters
+          )
+          identity_provider.provide_jwt_identity
+        end
+
+        def validate_restrictions(authentication_parameters)
+          initialize_validate_restrictions
+          @validate_resource_restrictions.call(
+            authenticator_name: authentication_parameters.authenticator_name,
+            service_id: authentication_parameters.service_id,
+            account: authentication_parameters.account,
+            role_name: authentication_parameters.jwt_identity,
+            constraints: @constraints,
+            authentication_request: @restriction_validator.new(
+              decoded_token: authentication_parameters.decoded_token
+            )
+          )
+        end
+
+        def validate_and_decode_token(authentication_parameters)
+          set_authentication_parameters(authentication_parameters)
+          initialize_validate_and_decode_token
+          @validate_and_decode_token.call(
+            authentication_parameters: authentication_parameters
+          )
+        end
+
+        private
+
+        def set_authentication_parameters(authentication_parameters)
+          @authentication_parameters = authentication_parameters
+        end
+
+        def initialize_validate_and_decode_token
+          initialize_validate_and_decode_token_dependencies
+          @validate_and_decode_token ||= @validate_and_decode_token_class.new(
+            fetch_signing_key: fetch_signing_key,
+            fetch_jwt_claims_to_validate: fetch_jwt_claims_to_validate
+          )
+        end
+
+        def initialize_validate_and_decode_token_dependencies
+          fetch_signing_key
+          fetch_jwt_claims_to_validate
+        end
```

Comment:
this method is redundant. the functions that it calls are memoized so you can just call them in the `initialize_validate_and_decode_token` method and remove the call to `initialize_validate_and_decode_token_dependencies`